pointcloud_boxcrop/src/pointcloud_boxcrop_node.cpp

```diff
@@ -0,0 +1,184 @@
+// Copyright 2025 Husarion sp. z o.o.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Eigen/Core>
+#include <pcl/conversions.h>
+#include <pcl/filters/crop_box.h>
+#include <pcl/point_cloud.h>
+#include <pcl/point_types.h>
+#include <rclcpp/rclcpp.hpp>
+
+#include <sensor_msgs/msg/point_cloud2.hpp>
+#include <tf2_geometry_msgs/tf2_geometry_msgs.hpp>
+
+#include "pointcloud_boxcrop/pointcloud_boxcrop_node.hpp"
+
+PointcloudBoxcropNode::PointcloudBoxcropNode()
+    : rclcpp::Node("pointcloud_boxcrop"), tf_buffer_(this->get_clock()),
+      tf_listener_(tf_buffer_) {
+  param_listener_ = std::make_shared<pointcloud_boxcrop_params::ParamListener>(
+      get_node_parameters_interface());
+
+  params_ = param_listener_->get_params();
+
+  pub_ = this->create_publisher<sensor_msgs::msg::PointCloud2>(
+      params_.output_topic, 10);
+  sub_ = this->create_subscription<sensor_msgs::msg::PointCloud2>(
+      params_.input_topic, 10,
+      std::bind(&PointcloudBoxcropNode::PointcloudCallback, this,
+                std::placeholders::_1));
+
+  if (params_.visualize_bounding_box) {
+    RCLCPP_INFO(this->get_logger(), "Bounding box visualization enabled.");
+    bbox_pub_ = this->create_publisher<vision_msgs::msg::BoundingBox3D>(
+        "~/filter_bounding_box", 10);
+  } else {
+    RCLCPP_INFO(this->get_logger(), "Bounding box visualization disabled.");
+  }
+
+  RCLCPP_INFO(this->get_logger(), "Initialized successfully.");
+}
+
+void PointcloudBoxcropNode::PointcloudCallback(
+    const sensor_msgs::msg::PointCloud2::SharedPtr msg) {
+  pcl::PointCloud<pcl::PointXYZ>::Ptr cloud(
+      new pcl::PointCloud<pcl::PointXYZ>());
+
+  std::string error_msg;
+  if (!tf_buffer_.canTransform(params_.target_frame, msg->header.frame_id,
+                               tf2::TimePointZero, &error_msg)) {
+    RCLCPP_WARN_STREAM(this->get_logger(),
+                       "Transform not found for frame: " << error_msg);
+    return;
+  }
+
+  geometry_msgs::msg::TransformStamped transform_stamped =
+      GetTransform(msg->header.frame_id);
```

Comment:
Hi Kuba, sorry for the confusion, but something was bothering me with this function and I noticed two things in the source code:
1. There's more than one exception, so lookupTransform might sometimes throw an exception despite canTransform (I'm not 100% sure, but it's better to be safe).
2. lookupTransform uses canTransform inside the body of the function, so it probably doesn't make sense to call it explicitly.

So, I suggest going back to try/catch but putting it in a  function so it's immediately clear that something could go wrong.`getTransform()` seems unnecessary, because in the callback you have to check whether to continue executing the callback.


```suggestion
  try {
    transform_stamped = tf_buffer_.lookupTransform(
        params_.target_frame, source_frame, tf2::TimePointZero);
  } catch (tf2::TransformException &ex) {
    RCLCPP_WARN_STREAM(this->get_logger(), "Could not transform "
                                               << params_.target_frame << " to "
                                               << source_frame << ": "
                                               << ex.what());
    return
  }
```